components/ai_chat/resources/page/components/main/index.tsx

```diff
@@ -85,6 +86,12 @@ function Main() {
         <ErrorConversationEnd />
       )
     }
+
+    if (apiHasError && currentError === mojom.APIError.GeneratedQuestionsRateLimitReached) {
+      currentErrorElement = (
+        <ErrorGeneratedQuestionsRateLimit />
+      )
+    }
   }
```

Comment:
can probably clean this up a bit now - e.g. wrap all of these conditions in a single `if (apiHasError)` block and even use a switch statement?

Reply:
refactored